File: online-judges/CodeJam/Qualification-Round-2010/A/main.cpp
/**
 * Problem: A. Bot Trust
 * Judge: CodeJam
 * Date: 22/11/2017
 * URL: https://code.google.com/codejam/contest/975485/dashboard#s=p0
 * Solved by: hopkins
 */

#include <stdio.h>

int main(int argc, char const *argv[])
{
    // freopen("A-small-practice.in", "r", stdin);
    // freopen("A-small-practice.sol", "w", stdout);
    // freopen("A-large-practice.in", "r", stdin);
    // freopen("A-large-practice.sol", "w", stdout);
    int tc;
    scanf("%d", &tc);
    for (int caso = 1; caso <= tc; ++caso) {
        int n, k;
        scanf("%d %d", &n, &k);

        int mask = (1 << n) - 1;

        printf("Case #%d: ", caso);
        if ((k & mask) == mask){
            printf("ON\n");
        } else {
            printf("OFF\n");
        }
    }
    return 0;
}

File: online-judges/CodeJam/Qualification-Round-2010/A/main.py
import sys

lines = sys.stdin.readlines()

for caso in range(1, len(lines)):
    [n, k] = map(int, lines[ caso ].split(' '))
    m = ((2**n) - 1)

    ans = 'OFF'
    if (k and m) == m:
        ans = 'ON'
    
    print('Case #' + str(caso) + ': ' + ans)